api/src/main/java/org/itsallcode/openfasttrace/api/core/LinkedSpecificationItem.java

```diff
@@ -281,23 +324,44 @@ public boolean isCoveredShallow()
         return this.getCoveredArtifactTypes().containsAll(this.getNeedsArtifactTypes());
     }
 
+    /**
+     * @return <code>true</code> if all needed attribute types are covered by approved items
+     */
+    public boolean isCoveredShallowWithApprovedItems()
+    {
+        return getStatus() == ItemStatus.APPROVED && this.getCoveredApprovedAttributeTypes()
+                .containsAll(this.getNeedsArtifactTypes());
+    }
+
     /**
      * Check if this item and all items providing coverage for it are covered.
-     * 
+     *
      * @return covered, uncovered or ring.
      */
     // [impl->dsn~tracing.deep-coverage~1]
     public DeepCoverageStatus getDeepCoverageStatus()
     {
         return getDeepCoverageStatusEndRecursionStartingAt(this.getId(),
-                DeepCoverageStatus.COVERED);
+                DeepCoverageStatus.COVERED, false);
+    }
+
+    public DeepCoverageStatus getDeepCoverageStatusOnlyAcceptApprovedItems()
+    {
+        return getDeepCoverageStatusEndRecursionStartingAt(this.getId(),
+                DeepCoverageStatus.COVERED, true);
     }
 
     // [impl->dsn~tracing.link-cycle~1]
     private DeepCoverageStatus getDeepCoverageStatusEndRecursionStartingAt(
-            final SpecificationItemId startId, final DeepCoverageStatus worstStatusSeen)
+            final SpecificationItemId startId, final DeepCoverageStatus worstStatusSeen,
+            final boolean onlyAcceptApprovedItemStatus)
     {
         DeepCoverageStatus status = worstStatusSeen;
+        if (onlyAcceptApprovedItemStatus && status == DeepCoverageStatus.COVERED && getStatus() != ItemStatus.APPROVED)
+        {
+            status = DeepCoverageStatus.UNCOVERED;
+        }
```

Comment:
Extract this `if` and give it a speaking name.

Reply:
DONE